### ISSUE TITLE
Recursive training on infinite stream of data

### ISSUE THREAD

**Issue by schlichtanders** [SciML/DiffEqFlux.jl#833]
Hello,

via the DiffEqFlux documentation I found [this paper](https://arxiv.org/abs/1907.07587) which demonstrates that Neural StochasticDifferentialEquations NeuralSDEs can be used to model financial data.

I am now looking for a way to learn such NeuralSDEs models iteratively/recursively, so that I can adapt to new incoming data efficiently.

The paper itself used a simple mean-squared error loss function, or something similar, for which you would need to store all historic data. Instead of relearning from the entire dataset every time, I rather want to update the previous fit in an efficient manner.

While I also learned about the math of SDEs in academia, I am more familiar with Bayesian probability modelling - and here the update logic is solved by using the previously found posterior as the new prior. Can something similar also be achieved for SDEs?

**Comment by ChrisRackauckas** [comment]
I don't see any potential issues with doing an online learning of SDEs, especially using a Bayesian approach using the previous posterior as a prior to the next one. This would just be a form of data assimilation on neural SDEs. It should just work. Please open an issue with a concrete feature request or bug report if it doesn't work as intended.
